test/testdrive/kafka-avro-sinks-doc-comments.td

```diff
@@ -82,6 +82,51 @@ $ schema-registry-verify schema-type=avro subject=testdrive-sink3-${testdrive.se
 $ schema-registry-verify schema-type=avro subject=testdrive-sink3-${testdrive.seed}-key
 {"type":"record","name":"row","doc":"comment on table t with a \\\\ \\","fields":[{"name":"c2","type":"string","doc":"doc on t.c2"}]}
 
+# Explain schema
+> EXPLAIN VALUE SCHEMA FOR CREATE SINK sink FROM t
```

Comment:
Just dropping in to say I think for ergonomics we should definitely allow omitting the sink name here. Doesn't seem too terrible. I'd just make the `name` field of `CreateSinkStatement` optional, and then in the planner `plan_create_sink` can error if the name is omitted, while `explain_schema` can happily carry on without a name.

Reply:
Er, I guess as written it's a little harder than that (need to plumb a bool to a shared helper indicating whether the create sink plan is allowed to omit the name or not), but still seems fundamentally tractable.

Reply:
I started with the changes for this but it's a bit of a noisy refactor with more testing and IMO would be better if it's a separate PR. I will do it as a follow up after this PR is merged unless you feel strongly against it.

(I think even if the sink name is required, the feature is still complete and usable with a fake name so the changes for EXPLAIN SCHEMA can still go out. Making the name optional is a nice to have on top of it.)

Reply:
> I will do it as a follow up after this PR is merged unless you feel strongly against it.

Sounds good.

Reply:
Another option here is to just plan it with a fake name e.g. using a hash or `find_available_name` on the catalog. Seems like it would be minimally invasive and not muddy either the parsing or planning of `CREATE SINK`.

Reply:
Oh, I like that idea.

Reply:
That would be nice! Since the catalog is not available during parsing, I think hash would be better. Let me try this out real quick.

Reply:
The name would need to be generated during planning, not parsing--i.e. we should only represent what users type in in the parser, so both of these would have something like `unwrap_or`...and then generate the name.

The "issue" with using a hash is that it might "break" the identifier rules by beginning with a number. At this point, we'd have already circumvented that issue which exists only in parsing but ¯\\\_(ツ)\_/¯

Reply:
Oh I see. Then we'll still need the name returned by the parser to be optional and the related refactoring. Will try this out in the follow up then.